pkg/spec/spec.go

```diff
@@ -347,21 +357,23 @@ func CreateConfigToOCISpec(config *CreateConfig) (*spec.Spec, error) { //nolint
 	configSpec.Mounts = supercedeUserMounts(volumeMounts, configSpec.Mounts)
 	//--mount
 	configSpec.Mounts = supercedeUserMounts(config.initFSMounts(), configSpec.Mounts)
-	if canAddResources {
-		// BLOCK IO
-		blkio, err := config.CreateBlockIO()
-		if err != nil {
-			return nil, errors.Wrapf(err, "error creating block io")
-		}
-		if blkio != nil {
-			configSpec.Linux.Resources.BlockIO = blkio
-		}
+	// BLOCK IO
+	blkio, err := config.CreateBlockIO()
+	if err != nil {
+		return nil, errors.Wrapf(err, "error creating block io")
+	}
+	if blkio != nil {
+		configSpec.Linux.Resources.BlockIO = blkio
+		addedResources = true
 	}
 
-	// If we cannot add resources be sure everything is cleared out
-	if !canAddResources {
+	if rootless.IsRootless() {
+		if addedResources {
+			return nil, errors.New("invalid configuration, cannot set resources with rootless containers")
+		}
```

Comment:
On the fence.  Would it be better to just do a check for IsRootless() and then return at each of the places where you set addResources to true rather than wait until you get down here ( and potentially do a bunch of other processing) to do that check?  I do like the addedResources variable name better than the old one if you decide to retain this logic.

Reply:
This isn't really right. I think this patch is over-complicating things -- `runc` already will error out if a resource request is not fulfillable (but there are cases where they could be). You just need to remove the `IsRootless` check above all of the OCI configuration setting, and just see whether `runc` returns an error or not. Having this big binary switch is something that we're working on fixing in `runc` -- let's not re-introduce it in `libpod`...

Reply:
before that, we need to improve how we report the errors from the OCI runtime (they are currently managed by conmon), if I drop the additional check here, I get a cryptic error from podman:
```
$ bin/podman run --rm -ti --memory=2G alpine echo hi
error reading container (probably exited) json message: EOF
```

Reply:
I would prefer the prettier message, but this is an ugly PR. Although I can not think of a better way to handle it.